media-sound/audacity/audacity-3.2.2.ebuild

```diff
@@ -0,0 +1,234 @@
+	alsa? ( media-libs/alsa-lib )
+	ffmpeg? ( media-video/ffmpeg )
+	flac? ( media-libs/flac:=[cxx] )
+	id3tag? ( media-libs/libid3tag:= )
+	lv2? (
+		dev-libs/serd
+		dev-libs/sord
+		>=media-libs/lilv-0.24.6-r2
+		media-libs/lv2
+		media-libs/sratom
+		media-libs/suil
+	)
+	mad? ( >=media-libs/libmad-0.15.1b )
+	mpg123? ( media-sound/mpg123 )
+	ogg? ( media-libs/libogg )
+	opus? ( media-libs/opus )
+	sbsms? ( media-libs/libsbsms )
+	telemetry? ( ${MY_NETDEPEND} )
+	audiocom? ( ${MY_NETDEPEND} )
+	twolame? ( media-sound/twolame )
+	vamp? ( media-libs/vamp-plugin-sdk )
+	vorbis? ( media-libs/libvorbis )
+	wavpack? ( media-sound/wavpack )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="app-arch/unzip
+	sys-devel/gettext
+	virtual/pkgconfig
+"
+
+REQUIRED_USE="
+	mad? ( !mpg123 )
+"
+
+PATCHES=(
+	# Equivalent to previous versions
+	"${FILESDIR}/${PN}-3.2.2-disable-ccache.patch"
+	# From Debian
+	"${FILESDIR}/${PN}-3.2.2-fix-rpaths.patch"
+
+	# Disables some header-based detection
+	"${FILESDIR}/${PN}-3.2.2-allow-overriding-alsa-jack.patch"
+
+	# Allows re-enabling telemetry
+	"${FILESDIR}/${PN}-3.2.2-remove-conan-threadpool.patch"
+	"${FILESDIR}/${PN}-3.2.2-allow-finding-rapidjson-pkgconfig.patch"
+)
+
+src_prepare() {
+	cmake_src_prepare
+
+	local header_subs="${S}/lib-src/header-substitutes"
+	cat <<-EOF >"${header_subs}/allegro.h" || die
+/* Hack the allegro.h header substitute to use system headers.  */
+#include <portsmf/allegro.h>
+EOF
+
+	# Keep condition in sync with has_networking in src_configure.
+	if use telemetry || use audiocom; then
+		mkdir -p "${S}/"/lib-src/threadpool/ThreadPool/ || die
+		cp "${DISTDIR}"/progschj-ThreadPool-"${MY_THREADPOOL_DATE}".h \
+		   "${S}"/lib-src/threadpool/ThreadPool/ThreadPool.h || die
+	fi
+}
+
+# NOTES:
+# I traced a bunch of dlopens via LD_AUDIT, and concluded that
+# dev-libs/{serd,sord}, media-libs/{opus,sratom}, media-sound/lame,
+# media-video/ffmpeg (yes), sys-libs/zlib are all used via dlopen.
+# FFmpeg in particular could even lose it's slot dependency, because
+# Audacity picks which version of the ABI it interacts with at runtime,
+# but I'm not sure how good of an idea that is.  Anyway - for this
+# release, the supported Lavc versions are 5[789], Lavu are 5[2567], and
+# Lavf are 5[789].  Check libraries/lib-ffmpeg-support/impl if you want
+# to check.  It's also useful to grep for wxDynamicLibrary and dlsym.
+
+src_configure() {
+	setup-wxwidgets
+
+	# Audio.com uploading and telemetry both require networking
+	# features.
+	local has_networking=off
+	if use audiocom; then
+		has_networking=on
+	fi
+
+	if use telemetry; then
+		has_networking=on
+		# Extracted from the official AppImage build.
+		local -x SENTRY_DSN_KEY="Temporarily omitted."
+		local -x SENTRY_HOST="Temporarily omitted."
+		local -x SENTRY_PROJECT="Temporarily omitted."
+		# This omits crash reporting, since Audacity fails to compile
+		# with our current breakpad, and upstream doesn't have our
+		# users' debuginfo anyway.
+	fi
```

Comment:
... so you're saying telemetry doesn't work, even if it's enabled? :) 

I guess it'd make sense to afterwards mask the use flag, open a bug explaining what needs to be done to get it working, and add that bug link to the mask message.

(let's try not to include any profile/* files in this PR so it won't become hard to merge due to conflicts)

Reply:
Telemetry does partially work, in that errors get sent to their sentry instance (as far as I could verify from tracing and sniffing TLS, I didn't mitmproxy yet).  Crash reporting specifically does not (and as I mentioned, I'm unsure if it even could)